### PR TITLE
[Merged by Bors] - chore: robustify proof in CategoryTheory.Limit.VanKampen

### PR DESCRIPTION
**Pull request by kim-em**
Defensive change preparing for https://github.com/leanprover/lean4/pull/4595.

### CODE REVIEW

File: Mathlib/CategoryTheory/Limits/VanKampen.lean [hunk 1]
```diff
@@ -570,9 +570,9 @@ theorem BinaryCofan.isVanKampen_mk {X Y : C} (c : BinaryCofan X Y)
     have he₁ : c'.inl = e₁.hom ≫ (cones f c.inl).fst := by simp [e₁]
     have he₂ : c'.inr = e₂.hom ≫ (cones f c.inr).fst := by simp [e₂]
     rw [he₁, he₂]
-    apply BinaryCofan.isColimitCompRightIso (BinaryCofan.mk _ _)
-    apply BinaryCofan.isColimitCompLeftIso (BinaryCofan.mk _ _)
-    exact h₂ f
+    exact
+      (BinaryCofan.mk _ _).isColimitCompRightIso e₂.hom
+        ((BinaryCofan.mk _ _).isColimitCompLeftIso e₁.hom (h₂ f))
```
Comment (grunweg) [inline Mathlib/CategoryTheory/Limits/VanKampen.lean:575]: Optional: I tend to write this as
```suggestion
    exact (BinaryCofan.mk _ _).isColimitCompRightIso e₂.hom
      ((BinaryCofan.mk _ _).isColimitCompLeftIso e₁.hom (h₂ f))

**Review by grunweg — COMMENTED** [review]
Looks reasonable on its own.
maintainer merge

File: Mathlib/CategoryTheory/Limits/VanKampen.lean (hunk 1 above)
Comment (mattrobball) [reply Mathlib/CategoryTheory/Limits/VanKampen.lean:575]: I mildly lean this way also but leave it up to you.

**Review by mattrobball — APPROVED** [review]

**Comment by mattrobball** [comment]
bors delegate+

**Comment by semorrison** [comment]
bors merge

**Pull request closed without merging**